src/optimizer/join_order_optimizer.cpp

```diff
@@ -32,6 +27,8 @@ bool JoinOrderOptimizer::ExtractBindings(Expression &expression, unordered_set<i
 		D_ASSERT(colref.binding.table_index != DConstants::INVALID_INDEX);
 		// map the base table index to the relation index used by the JoinOrderOptimizer
 		D_ASSERT(relation_mapping.find(colref.binding.table_index) != relation_mapping.end());
+		cardinality_estimator.relation_to_columns[relation_mapping[colref.binding.table_index]].insert(
```

Comment:
Can we make a function in `CardinalityEstimator` for this? It's not really clear what this code does when reading it.

Something like this, or a more descriptive name that makes more sense to you:
```c++
cardinality_estimator.AddColumnToRelation(...);
```

Reply:
There's a few more instances in this file where you add something to a map or other data structure that belongs to the `CardinalityEstimator`.

Touching the data structures that belong to `CardinalityEstimator` from outside of the `cardinality_estimator.*pp` files should be done sparingly.

The loop at line 212 in this file (under `ExtractJoinRelations`), for example, adds some key/value pairs to a map in `CardinalityEstimator`. In those cases, the whole loop should probably be moved to the implementation file `cardinality_estimator.cpp`, as this has nothing to do with dynamic programming.

If it's some kind of initialisation like in `ExtractJoinRelations`, we prefer having a single call to some function in `CardinalityEstimator`, rather than having a few lines here and there that do some stuff related to cardinality estimation in our functions that belong to our DP implementation.